### PR TITLE
[Fix](variant) ignore serialization of nothing type

### PR DESCRIPTION
**Pull request by eldenmoon**
1. fix variant should not serialize nothing type
2. fix unstable cases
Issue Number: close #xxx

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/data_types/data_type_object.cpp [hunk 1]
```diff
@@ -113,6 +119,8 @@ char* DataTypeObject::serialize(const IColumn& column, char* buf, int be_exec_ve
         // 2.2 serialize subcolumn
         buf = type->serialize(entry->data.get_finalized_column(), buf, be_exec_version);
     }
+    // serialize num of subcolumns
+    *reinterpret_cast<uint32_t*>(size_pos) = num_of_columns;
```
Comment (xiaokang) [inline be/src/vec/data_types/data_type_object.cpp:123]: variable size_pos is only used once and can be elimilated.

Comment (eldenmoon) [reply be/src/vec/data_types/data_type_object.cpp:123]: no， it's need to be recorded, and assigned then

**Review by xiaokang — APPROVED** [review]

**Review by qidaye — APPROVED** [review]

**Pull request merged**